mpire/tqdm_utils.py

```diff
@@ -112,30 +110,36 @@ def start_manager(cls) -> bool:
             cls.MANAGER.start()
         cls.MANAGER_STARTED.set()
 
-        # Set host so other processes know where to connect to. On Linux, since Python 3.9, address is a bytes object
-        # and is prefixed by a null byte which needs to be removed (null byte doesn't work with Array). Before 3.9, this
-        # was a string, which we need to transform to bytes
-        if RUNNING_WINDOWS or (sys.version_info[0] == 3 and sys.version_info[1] < 9):
-            cls.MANAGER_HOST.value = cls.MANAGER.address.encode()
-        else:
-            cls.MANAGER_HOST.value = cls.MANAGER.address[1:]
+        # Set host so other processes know where to connect to. On some systems and Python versions address is a bytes
+        # object, on others it's a string. On some it's also prefixed by a null byte which needs to be removed (null
+        # byte doesn't work with Array).
+        address = cls.MANAGER.address
+        if isinstance(address, str):
+            address = address.encode()
+        if address[0] == 0:
+            address = address[1:]
+        cls.MANAGER_HOST.value = address
 
         return True
 
     def connect_to_manager(self) -> None:
         """
         Connect to the tqdm manager
         """
-        # Connect to a server. On Linux, since Python 3.9, the address is prefixed by a null byte (which was stripped
-        # when setting the host value, due to restrictions in Array). Address needs to be a string.
-        if RUNNING_WINDOWS or (sys.version_info[0] == 3 and sys.version_info[1] < 9):
-            address = self.MANAGER_HOST.value.decode()
-        else:
-            address = f"\x00{self.MANAGER_HOST.value.decode()}"
-        self.MANAGER = SyncManager(address=address, authkey=b'mpire_tqdm')
-        self.MANAGER.register('get_tqdm_lock')
-        self.MANAGER.register('get_tqdm_position_register')
-        self.MANAGER.connect()
+        # Connect to a server. On some systems and Python versions the address is prefixed by a null byte (which was
+        # stripped when setting the host value, due to restrictions in Array). Address needs to be a string.
+        address = self.MANAGER_HOST.value.decode()
+        try:
+            self.MANAGER = SyncManager(address=address, authkey=b'mpire_tqdm')
+            self.MANAGER.register('get_tqdm_lock')
+            self.MANAGER.register('get_tqdm_position_register')
+            self.MANAGER.connect()
+        except FileNotFoundError:
```

Comment:
Why would it throw a file not found error? Does it try to save some stuff to disk but can't because the listening address is wrong? There then is probably no other way to check if the address is legit other then just letting the exception occur?

Reply:
That's how Managers work in Python. They use sockets for communication and addresses of sockets are bound to a file system node (see https://docs.python.org/3/library/socket.html#socket-families, first bullet point). If you try to connect to one that doesn't exist (because the address misses the null byte) then you get a FileNotFoundError.